src/app/auth/login/login.component.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { RouterTestingModule } from "@angular/router/testing";
+
+import { LoginComponent } from './login.component';
+import { Observable } from 'rxjs';
+
+describe('LoginComponent', () => {
+  let usuarioService:UsuarioService
+  let fb:FormBuilder
+  let component: LoginComponent;
+  let fixture: ComponentFixture<LoginComponent>;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [ LoginComponent ],
+      
+      imports: [
+        RouterTestingModule,
+        HttpClientTestingModule,
+        ReactiveFormsModule
+      ]
+    })
+    .compileComponents();
+    fixture = TestBed.createComponent(LoginComponent);
+    component = fixture.componentInstance;
+    usuarioService = TestBed.inject(UsuarioService)
+    fb = TestBed.inject(FormBuilder)
+    fixture.detectChanges();
+  });
+
+  describe('Formulario', () => {
+    it('Formulario: Iniciado correctamente', () => {
+      let email = component.loginForm.controls['email']
+      let password = component.loginForm.controls['password']
+      
+      expect(email.value).toEqual('')
+      expect(password.value).toEqual('')
+      expect(component.loginForm.valid).toBeFalsy()
+    })
+  
+    it('Formulario: Validado', () => {
+      let email = component.loginForm.controls['email']
+      let password = component.loginForm.controls['password']
+      email.setValue('jescoevas@hotmail.com')
+      password.setValue('1234')
+      expect(email.value).toEqual('jescoevas@hotmail.com')
+      expect(password.value).toEqual('1234')
+      expect(component.loginForm.valid).toBeTruthy()
+    })
+  })
+
+    xdescribe('Login', () => {
+    it('Login: Datos enviados', () => {
+      let email = component.loginForm.controls['email']
+      let password = component.loginForm.controls['password']
+      
+      email.setValue('jescoevas@hotmail.com')
+      password.setValue('1234')
+      spyOn(usuarioService, 'login')
+      component.login()
+      expect(usuarioService.login).toHaveBeenCalledWith(component.loginForm.value)
+    })
+    
+    it('Login: Datos no enviados por password vacía', () => {
+      let email = component.loginForm.controls['email']
+      let password = component.loginForm.controls['password']
+      email.setValue('jescoevas@hotmail.com')
+      password.setValue('')
+      spyOn(usuarioService, 'login')
+      component.login()
+      expect(usuarioService.login).not.toHaveBeenCalled()
+    })
+
+    it('Login: Botón de submit funciona correctamente', () => {
+      let email = component.loginForm.controls['email']
+      let password = component.loginForm.controls['password']
+      email.setValue('jescoevas@hotmail.com')
+      password.setValue('1234')
+      spyOn(usuarioService, 'login')
+      const button = fixture.debugElement.nativeElement.querySelector('#submit');
+      button.click();
+      expect(usuarioService.login).toHaveBeenCalled()
+    })
+
+    it('Login: Login exitoso', () => {
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ComfyStream/ComfyStream-Frontend/pullRequest?prid=7138913)
```suggestion
    it("Login: Login exitoso", () => {
```